contributing-to-open-source.md

```diff
@@ -0,0 +1,249 @@
+some good practices when working with forks.
+
+
+## Committing from non-Visma email
+
+If you don't want to expose your Visma email address in the commits,
+or want to have the commits associated with your personal email
+instead, you have to change the per-repository settings:
+
+    cd some-open-source-project
+    git config user.email xzyzzy@example.com
+
+SourceTree and other graphical interfaces also let you change this.
+If you forget to do this and create some commits, you can [change the
+author information on them afterwards with an interactive
+rebase](https://stackoverflow.com/questions/3042437/how-to-change-the-commit-author-for-one-specific-commit#3042512).
+
+If you already have pushed the commits to your forked repository, you
+will have to do a forced push to replace them.
+
+
+# Improving documentation
+
+There are not that many people who enjoy writing documentation for
+free, so open source can be lacking in documention.  If you find
+errors or missing documentation and like to patch the gap, it will be
+much appreciated by the maintainers.
+
+Most projects maintain the documentation as part of the code
+repository, either as markdown files or inline code documentation
+comments.  Submitting fixes to documentation thus follows the same
+process as submitting a bug fix, but you don't have to create an issue
+beforehand and can submit the PR directly.
+
+Github has good support for editing markdown files directly on the
+website, so you don't have to clone your forked repository to your
+laptop but can work directly in the web GUI if you want.
+
+
+# Implementing new functionality
+
+If you have an idea for improvements or extensions that is general
+enough to support both you and other users, start by submitting an
+issue explaining it and outlining how you think that you can implement
+it.  Expect discussion in the issue, as the project maintainers can
+have their own plans which means that the change is not a good fit as
+suggested, or suggest a better way of how to implement it.
+
+If the maintainers agree on the idea, go ahead and implement it
+following the same process as when submitting a bug.  Don't forget to
+update relevant documentation too.
+
+The new functionality cannot contain any Visma IP, as the code will
+become open source too.  Get your managers approval before submitting
+any code.  The contributor's agreement may also require you to assign
+the copyright to the project.
+
+
+# Reviewing and testing pull requests
+
+Once you have some significant code in a project or really care about
+its future direction, you may want to start to monitor the pull
+requests that come in and review the ones that touch on your
+interests.  QAs can help test PRs too.
+
+
+# Maintaining open source projects
+
+If you start being visible in projects by submitting code, discussing
+issues and reviewing PRs, you may be asked by to become a maintainer.
+This typically comes with push rights to the main repository and clout
```

Comment:
What is "clout"?

Reply:
https://www.merriam-webster.com/dictionary/clout meaning 4 :)

Reply:
"influence over" is a better phrase, in any case.